test/widgets/profile_test.dart

```diff
@@ -0,0 +1,269 @@
+    displayInProfileSummary: displayInProfileSummary ?? true,
+    fieldData: fieldData ?? '',
+    type: type,
+  );
+}
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  group('ProfilePage', () {
+    testWidgets('page builds; profile page renders', (WidgetTester tester) async {
+      final user = eg.user(userId: 1, fullName: 'test user');
+
+      await setupPage(tester, users:[user], pageUserId: user.userId);
+
+      check(because: 'find user avatar', find.byType(Avatar).evaluate()).length.equals(1);
+      check(because: 'find user name', find.text('test user').evaluate()).isNotEmpty();
+    });
+
+    testWidgets('page builds; profile page renders with profileData', (WidgetTester tester) async {
+      await setupPage(tester,
+        users: [
+          eg.user(userId: 1, profileData: {
+            0: ProfileFieldUserData(value: 'shortTextValue'),
+            1: ProfileFieldUserData(value: 'longTextValue'),
+            2: ProfileFieldUserData(value: 'x'),
+            3: ProfileFieldUserData(value: 'dateValue'),
+            4: ProfileFieldUserData(value: 'http://example/linkValue'),
+            5: ProfileFieldUserData(value: '[2]'),
+            6: ProfileFieldUserData(value: 'externalValue'),
+            7: ProfileFieldUserData(value: 'pronounsValue'),
+          }),
+          eg.user(userId: 2),
+        ],
+        pageUserId: 1,
+        customProfileFields: [
+          mkCustomProfileField(0, CustomProfileFieldType.shortText),
+          mkCustomProfileField(1, CustomProfileFieldType.longText),
+          mkCustomProfileField(2, CustomProfileFieldType.choice, fieldData: '{"x": {"text": "choiceValue", "order": "1"}}'),
+          mkCustomProfileField(3, CustomProfileFieldType.date),
+          mkCustomProfileField(4, CustomProfileFieldType.link),
+          mkCustomProfileField(5, CustomProfileFieldType.user),
+          mkCustomProfileField(6, CustomProfileFieldType.externalAccount, fieldData: '{"subtype": "external1"}'),
+          mkCustomProfileField(7, CustomProfileFieldType.pronouns),
+        ], realmDefaultExternalAccounts: {
+          'external1': RealmDefaultExternalAccount(
+            name: 'external1',
+            text: '',
+            hint: '',
+            urlPattern: 'https://example/%(username)s')});
+
+      final testCases = [
+        (find.text('field0'), find.text('shortTextValue'), CustomProfileFieldType.shortText),
+        (find.text('field1'), find.text('longTextValue'), CustomProfileFieldType.longText),
+        (find.text('field2'), find.text('choiceValue'), CustomProfileFieldType.choice),
+        (find.text('field3'), find.text('dateValue'), CustomProfileFieldType.date),
+        (find.text('field4'), find.text('http://example/linkValue'), CustomProfileFieldType.link),
+        (find.text('field5'), find.byType(Avatar), CustomProfileFieldType.user),
+        (find.text('field6'), find.text('http://example/externalValue'), CustomProfileFieldType.externalAccount),
```

Comment:
Hmm, this doesn't seem like the right behavior — the text should be just "externalValue".

Your screenshot in the PR description seems to show the right behavior, though.

Hmmm and the tests still pass if I change this expected result:
```diff
-        (find.text('field6'), find.text('http://example/externalValue'), CustomProfileFieldType.externalAccount),
+        (find.text('field6'), find.text('asdf'), CustomProfileFieldType.externalAccount),
```

Seems like there's a bug in the test somewhere.